server/tests/article.test.js

```diff
@@ -0,0 +1,119 @@
+import 'babel-polyfill'
+import path from 'path'
+import { GraphQLError } from 'graphql'
+import { ApolloServer, gql } from 'apollo-server'
+import { describe, expect, jest, test } from '@jest/globals'
+import { importSchema } from 'graphql-import'
+import { createTestClient } from 'apollo-server-testing'
+import resolvers from '../resolvers'
+import { databaseApi, searchApi } from '../endpoints'
+import { articles } from './mocks'
+
+const typeDefs = importSchema(path.join(__dirname, '../typedefs/index.graphql'))
+const server = new ApolloServer({ typeDefs, resolvers })
+
+const tests = [
+  {
+    name: 'Normal request with a valid search',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    id
+                    number
+                    content
+                    keywords
+                    minutesToRead
+                }
+            }
+        `,
+    variables: { search: 'Es obligatorio usar casco con bicicleta?' },
+    response: { data: { articles: ['1'] } },
+    expected: { articles: [articles[0]] }
+  },
+  {
+    name: 'Specific attributes to select',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    number
+                    content
+                }
+            }
+        `,
+    variables: { search: 'Es obligatorio usar casco con bicicleta?' },
+    response: { data: { articles: ['1'] } },
+    expected: { articles: [articles[0]].map(({ number, content }) => ({ number, content })) }
+  },
+  {
+    name: 'None match from request',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    id
+                    number
+                    content
+                    keywords
+                    minutesToRead
+                }
+            }
+        `,
+    variables: { search: 'A very thorough search that won\'t return anything...' },
+    response: { data: { articles: [] } },
+    expected: { articles: [] }
+  },
+  {
+    name: 'Search error',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    id
+                }
+            }
+        `,
+    variables: { search: '' },
+    response: { error: { message: 'An error message', trace: 'Its stack trace' } },
+    expected: [new GraphQLError(JSON.stringify({ message: 'An error message', trace: 'Its stack trace' }))]
+  },
+  {
+    name: 'Article in database does not exists',
+    query: gql`
+            query articles($search: String!) {
+                articles(search: $search) {
+                    id
+                }
+            }
+        `,
+    variables: { search: '' },
+    response: { data: { articles: ['20'] } },
+    expected: [new GraphQLError(JSON.stringify({ error: 'No article matches such ID', code: 404 }))]
```

Comment:
How is this handled on the frontend?

Reply:
Although this updates are expected for next PR, I was thinking of showing a popup containing the error data.

Reply:
I think this specific error has no need to be displayed, if a result id matches no article just don't display that ID